pkg/apis/security/v1beta1/types.go

```diff
@@ -0,0 +1,140 @@
+// allowed or denied which exactly match the specified ports and protocol.
+type Rule struct {
+	// Action specifies the action to be applied on the rule. Defaults to
+	// ALLOW action.
+	// +optional
+	Action *RuleAction `json:"action"`
+	// Set of port and protocol allowed/denied by the rule. If this field is unset
+	// or empty, this rule matches all ports.
+	// +optional
+	Ports []NetworkPolicyPort `json:"ports"`
+	// Rule is matched if traffic originates from workloads selected by
+	// this field. If this field is empty, this rule matches all sources.
+	// +optional
+	From []NetworkPolicyPeer `json:"from"`
+	// Rule is matched if traffic is intended for workloads selected by
+	// this field. If this field is empty or missing, this rule matches all
+	// destinations.
+	// +optional
+	To []NetworkPolicyPeer `json:"to"`
+}
+
+// NetworkPolicyPeer describes the grouping selector of workloads.
+type NetworkPolicyPeer struct {
+	// IPBlock describes the IPAddresses/IPBlocks that is matched in to/from.
+	// IPBlock cannot be set as part of the AppliedTo field
+	// Cannot be set with any other selector.
+	// +optional
+	IPBlock *IPBlock `json:"ipBlock,omitempty"`
+	// Select Pods from all Namespaces as workloads in AppliedTo/To/From
+	// fields. If set with NamespaceSelector, Pods are matched from Namespaces
+	// matched by the NamespaceSelector.
+	// Cannot be set with any other selector except NamespaceSelector.
+	// +optional
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
+	// Select all Pods from Namespaces matched by this selector, as
+	// workloads in AppliedTo/To/From fields. If set with PodSelector,
+	// Pods are matched from Namespaces matched by the NamespaceSelector.
+	// Cannot be set with any other selector except PodSelector.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+}
+
+// IPBlock describes a particular CIDR (Ex. "192.168.1.1/24") that is allowed
+// or denied to/from the workloads matched by a Spec.AppliedTo.
+type IPBlock struct {
+	// CIDR is a string representing the IP Block
+	// Valid examples are "192.168.1.1/24".
+	CIDR string `json:"cidr"`
+}
+
+// NetworkPolicyPort describes the port and protocol to match in a rule.
+type NetworkPolicyPort struct {
+	// The protocol (TCP, UDP, or SCTP) which traffic must match.
+	// If not specified, this field defaults to TCP.
+	// +optional
+	Protocol *v1.Protocol `json:"protocol"`
+	// The port on the given protocol. This can either be a numerical
+	// or named port on a Pod. If this field is not provided, this
+	// matches all port names and numbers.
+	// TODO: extend it to include Port Range.
+	// +optional
+	Port *intstr.IntOrString `json:"port"`
+}
+
+// RuleAction describes the action to be applied on traffic matching a rule.
+type RuleAction string
+
+const (
+	// RuleActionAllow describes that rule matching traffic must be allowed.
```

Comment:
s/rule matching/rule-matching
or better IMO "traffic matching the rule"

same below

Reply:
done